pkg/sinks/clickhouse.go

```diff
@@ -0,0 +1,274 @@
+package sinks
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/resmoio/kubernetes-event-exporter/pkg/kube"
+	"github.com/rs/zerolog/log"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type ClickHouseConfig struct {
+	Hosts             []string `yaml:"hosts"`
+	Username          string   `yaml:"username"`
+	Password          string   `yaml:"password"`
+	Database          string   `yaml:"database"`
+	TableName         string   `yaml:"tableName"`
+	UseHttp           bool     `yaml:"useHttp"`
+	TLS               *TLS     `yaml:"tls"`
+	CreateTable       bool     `yaml:"createTable"`
+	TableEngine       string   `yaml:"tableEngine"`
+	Compress          string   `yaml:"compress"`
+	MaxIdleConns      *int     `yaml:"maxIdleConns"`
+	MaxOpenConns      *int     `yaml:"maxOpenConns"`
+	ConnMaxLifetimeMs *int     `yaml:"connMaxLifetimeMs"`
+	ConnMaxIdleTimeMs *int     `yaml:"connMaxIdleTimeMs"`
+}
+
+const (
+	defaultTableEngine = "MergeTree"
+	ddlFmt             = `CREATE TABLE %s (
+	KubeClusterName LowCardinality(String),
+	Reason LowCardinality(String),
+	Message String CODEC(ZSTD),
+	SourceComponent LowCardinality(Nullable(String)),
+	SourceHost Nullable(String),
+	FirstTimestamp DateTime,
+	LastTimestamp DateTime,
+	Count UInt32,
+	Type LowCardinality(String),
+	Kind LowCardinality(String),
+	Namespace LowCardinality(String),
+	Name LowCardinality(String),
+	UID UUID,
+	ResourceVersion String,
+	FieldPath Nullable(String),
+	Labels Map(String, String),
+	Annotations Map(String, String),
+	Deleted Bool,
+	OwnerReferenceAPIVersions Array(LowCardinality(String)),
+	OwnerReferenceKinds Array(LowCardinality(String)),
+	OwnerReferenceNames Array(LowCardinality(String)),
+	OwnerReferenceUIDs Array(UUID),
+	ReportingComponent LowCardinality(Nullable(String)),
+	ReportingInstance Nullable(String),
+) ENGINE = %s
+ORDER BY (Namespace, Name, LastTimestamp)
```

Comment:
Name, Namespace, LastTimestamp 

Order by key should be highest cardinality to least (left to right), with Timestamp at the right